clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/security/KubernetesV2Credentials.java

```diff
@@ -611,24 +611,22 @@ public void patch(
       String action, List<KubernetesKind> kinds, String namespace, Supplier<T> op) {
     Map<String, String> tags = new HashMap<>();
     tags.put("action", action);
-    if (kinds.size() == 1) {
-      tags.put("kind", kinds.get(0).toString());
-    } else {
-      tags.put(
-          "kinds", kinds.stream().map(KubernetesKind::toString).collect(Collectors.joining(",")));
-    }
+    tags.put(
+        "kinds",
+        kinds.stream().map(KubernetesKind::toString).sorted().collect(Collectors.joining(",")));
     tags.put("account", accountName);
     tags.put("namespace", StringUtils.isEmpty(namespace) ? "none" : namespace);
     tags.put("success", "true");
     long startTime = clock.monotonicTime();
     try {
       return op.get();
     } catch (KubectlException e) {
-      // TODO(ezimanyi): We should be setting success to false here; this does not feel like success
+      tags.put("success", "false");
+      tags.put("reason", e.getClass().getSimpleName());
       throw e;
     } catch (Exception e) {
       tags.put("success", "false");
-      tags.put("reason", e.getClass().getSimpleName() + ": " + e.getMessage());
+      tags.put("reason", e.getClass().getSimpleName());
       throw new KubectlJobExecutor.KubectlException(
           "Failure running " + action + " on " + kinds + ": " + e.getMessage(), e);
```

Comment:
Why do we only wrap non-KubectlException exceptions' messages in this custom "Failure running" message and just re-throw KubectlExceptions with their message as-is?

Reply:
To be honest, I also find this a bit confusing and am not 100% sure what would be best here.

I *think* the intent is to ensure that all exceptions leaving this function are a `KubectlException` to make it easier to catch.  On the one hand, that's good as it means we're throwing an exception appropriate to the level of abstraction, so callers will see that a call to `kubectl` (and which call) failed rather than some low-level exception.

I think the presumption is probably that if we already have a `KubectlException` it already has information at the right level of abstraction (and it does largely seem to as it usually contains information about the `kubectl` call that failed).  But it does feel like we should potentially be cleaner about where this exception is thrown and maybe:
* Throw a `KubectlException` in `KubectlJobExecutor` with details about the `kubectl` call that failed
* Wrap *any* exception in a `KubernetesAPIException` in this method providing details on the api operation that failed

I think this will probably be necessary for consistency when we work on using a client library, as `KubectlException` will be an implementation detail of the `kubectl` exception, which is why I didn't overly refactor it here.  In my follow-up PR to make `replace` an upsert, I will slightly improve this, but I think that the real improvement will come when we move to a client library.  (And I definitely want to talk with you in detail about what the exceptions should be as part of that work.)

Reply:
👍 👍 Thank you for the paragraphs. Even more excited for the client library migration now!